### ISSUE TITLE
Compiler hangs when I include Gadfly

### ISSUE THREAD

**Issue by ViralBShah** [JuliaLang/julia#22569]
Continuing from #22334 I am using a more complex script into a shared library. The only addition here is including Gadfly, without which it builds fine - but hangs when I add Gadfly.

```
module mystuff

using Gadfly
using Distributions
f() = rand(Normal())

Base.@ccallable Float64 function julia_main()
   println("hello, world")
   @show f()
end

end

#using mystuff
#mystuff.julia_main()
```

The compilation counter hangs, and happens unpredictably. It makes a different amount of progress on every run.

**Comment by ViralBShah** [comment]
Would having a smaller test case than Gadfly help here? Perhaps @mdpradeep can help narrow it down.

**Comment by vtjnash** [comment]
No, having the whole package to exercise assorted code paths is valuable

**Comment by vtjnash** [comment]
fixed by #22697 (but forgot to reference this correctly for auto-close)
